libraries/resource_hab_service.rb

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright:: Copyright 2016, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource/service"
+
+class Chef
+  class Resource
+    class HabSup < Chef::Resource::Service
+      include HabitatCookbook::Helpers
+
+      resource_name :hab_service
+      provides :hab_service
```

Comment:
same comment here as in the provider:

```ruby
provides :hab_service_systemd
provides :hab_service do |node|
  Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
end
```

also the file and the class needs to be renamed to `resource_hab_service_systemd.rb` and `HabServiceSystemd`